tests/test_snowflake_common.py

```diff
@@ -386,14 +388,19 @@ def test__describe_api_didnt_describe(connect, mocker):
 
 @patch(
     'pandas.DataFrame.from_dict',
-    side_effect=[pd.DataFrame(data_result_all), pd.DataFrame({'TOTAL_ROWS': 20}, index=[0])],
+    side_effect=[
+        None,
+        None,
+        pd.DataFrame(data_result_all),
+        pd.DataFrame({'TOTAL_ROWS': 20}, index=[0]),
+    ],
 )
-def test_retrieve_data_with_row_count_limit_in_query(fetchmany, snowflake_datasource, mocker):
+@patch('snowflake.connector.connect', return_value=snowflake.connector.SnowflakeConnection)
+def test_retrieve_data_with_row_count_limit_in_query(connect, fetchmany, snowflake_datasource):
     snowflake_datasource.query = 'select name from favourite_drinks limit 10;'
-    connect = mocker.MagicMock()
     sc = SnowflakeCommon()
     sc.retrieve_data(connect, snowflake_datasource, get_row_count=True)
-    assert fetchmany.call_count == 2
+    assert fetchmany.call_count == 4
```

Comment:
```suggestion
    assert fetchmany.call_count == 4  # 2 for each connection
```
Je te propose un ptit commentaire pour aider à comprendre.
Et d'ailleurs tes mocks ont un souci non? Tu lies le 1er au 2e a priori

Reply:
Je suis d'accord pour le commentaire
Je ne suis pas sûr de comprendre le point sur les Mock ?